backend/src/routes/auth.ts

```diff
@@ -160,4 +164,122 @@ export const authRoute = new Elysia({ prefix: '/auth' })
             status: 'success',
             message: `${user.email} has been verified`,
         };
+    })
+    .post('/sign-in', async ({ request, cookie, set }) => {
+        const formData = await request.formData();
+        const validEmaillPass = signInFormSchema.safeParse({
+            email: formData.get('email'),
+            password: formData.get('password'),
+        });
+
+        if (!validEmaillPass.success) {
+            return {
+                status: 'error',
+                errors: validEmaillPass.error.flatten().fieldErrors,
+            };
+        }
+        
+        const { email, password } = validEmaillPass.data;
+        
+        let user_id = '';
+        
+        try {
+            const queryAuthUserData = await sql`
+            SELECT id, hashed_password
+            FROM auth_user
+            WHERE email = ${email}
+            `;
+            
+            const queriedHashedPassword = queryAuthUserData[0].hashed_password;
+            user_id = queryAuthUserData[0].id;
+            
+            const isPasswordMatch = await new Argon2id().verify(
+                queriedHashedPassword,
+                password,
+                );
+            
+            set.status = 401;
+            if( queryAuthUserData.length === 0 ) {
+                return {
+                    status: 'error',
+                    message: 'email or password is incorrect',
+                };
+            }
+                
+            set.status = 401;
+            if (!isPasswordMatch) {
+                return {
+                    status: 'error',
+                    message: 'email or password is incorrect',
+                };
+            }
+        } catch (error) {
+            set.status = 404;
+            return {
+                status: 'error',
+                message : 'An error occurred, please try again later'
+            };
+        }
```

Comment:
For database error, this would be a 500 error code.
But for generic error handler, I have already added global error handler, so this won't be neccesary

Reply:
So I have to remove the `error catch` right